core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BoundedRangeFileInputStream.java

```diff
@@ -62,12 +62,15 @@ public BoundedRangeFileInputStream(FSDataInputStream in, long offset, long lengt
 
   @Override
   public int available() throws IOException {
-    int avail = in.available();
-    if (pos + avail > end) {
-      avail = (int) (end - pos);
-    }
+    final FSDataInputStream inLocal = in;
+    synchronized (inLocal) {
+      int avail = inLocal.available();
```

Comment:
Why? DFS won't even do that for you. they'll take the file length - pos, why couldn't the same liberty be used in this case? Accumulo is inherently coupled with DFS why even attempt to promulgate the idea that the InputStream API should be adhered?